code/game/gamemodes/cult/structures/pylon.dm

```diff
@@ -257,85 +250,74 @@
 
 //Called every process in turret mode, and also by chaining spawns
 /obj/structure/cult/pylon/proc/handle_firing()
-
-	if ((world.time < next_shot) || isbroken)
+	if((world.time < next_shot) || isbroken)
 		return
 		//Not ready to fire yet
 
-
 	var/list/stuffcache
 	//Stuffcache holds a list of things found by a dview call, so we only need to do it once per proc.
 
-
-	if (target && (target.stat != DEAD))
-		if (target.loc == last_target_loc)
+	if(target && target.stat != DEAD)
+		if(target.loc == last_target_loc)
 		//To minimise expensive DVIEW calls, if the target hasnt moved since the last shot we'll assume they're still in sight
 		//This could cause problems in some edge cases (like lowering firelocks or shutters without moving)
 		//But it has a major performance gain that makes it worth it
 			fire_at(target)
 			return
 		else
 			stuffcache = mobs_in_view(9, src)
-			//for (var/turf/T in stuffcache)
-				//new /obj/effect/testtrans(T)
-			if ((target in stuffcache) && isInSight(src, target))
-				for (var/mob/m in stuffcache)
+			if((target in stuffcache) && isInSight(src, target))
 				fire_at(target)
 				return
 			else
 				target = null
 
 	//We may have already populated stuffcache this run, dont repeat work
-	if (!stuffcache)
+	if(!stuffcache)
 		stuffcache = mobs_in_view(9, src)
 
 	target = null //Either we lost a target or lack one
-	//for (var/turf/T in stuffcache)
-		//new /obj/effect/testtrans(T)
 
-	for (var/mob/living/L in stuffcache)
-		if (L == sacrificer)
+	for(var/mob/living/L in stuffcache)
+		if(L == sacrificer) //A pylon won't shoot the person who created it, regardless of cult status.
+			continue		 //This is mainly for xenoarch antags
+		if(L.stat == DEAD)  //No point shooting at corpses
 			continue
-			//A pylon won't shoot the person who created it, regardless of cult status.
-			//This is mainly for xenoarch antags
-
-		if (L.stat == DEAD)
-			continue
-			//No point shooting at corpses
-
-		if (iscult(L))
+		if(iscult(L))		 //Pylon wont shoot at cultists or constructs
 			continue
-			//Pylon wont shoot at cultists or constructs
-
-		if (!isInSight(src, L))
+		if(!isInSight(src, L))
 			continue
-
+		if(istype(L, /mob/living/heavy_vehicle))
```

Comment:
```suggestion
		if(ismech(L))
```

Reply:
There's a _lot_ of other defines, helpers and such you could correct in this file if you want to.